.github/workflows/dev.yml

```diff
@@ -168,21 +168,7 @@ jobs:
       - id: "build_each_commit"
         name: "build each commit"
         run: |
-          # Run a simple build for each separate commit (for "pull_request")
-          # or for the HEAD of the branch (other events).
-          set -eu -o pipefail
-          COMMITS=${{ github.sha }}
-          if [[ "${{ github.event_name == 'pull_request' }}" == "true" ]]; then
-            # Get all commits from Pull Request, in chronological order
-            COMMITS=$(git rev-list --reverse ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }})
-          fi
-          for commit in $COMMITS ; do
-            git checkout $commit || exit 1
-            printf "::group::Build commit %s\n" "$(git log --oneline --no-decorate -n 1)"
-            (just debug_justfile="${{matrix.debug_justfile}}" cargo +${{matrix.rust.version}} build --locked --profile=dev --target=x86_64-unknown-linux-gnu) || exit 1
-            printf "::endgroup::\n"
-          done
-          printf "::notice::HEAD remains at %s\n" "$(git log --oneline --no-decorate -n 1)"
+          just debug_justfile="${{matrix.debug_justfile}}" rust="${{matrix.rust.version}}" build-sweep
```

Comment:
You didn't check that one in CI:

![image](https://github.com/user-attachments/assets/31f73e91-531f-4343-8efe-9cd236ea7e35)

I'm also unsure `main` is defined when pulling the files with the `checkout` GitHub Action, you may have to provide the explicit commits as arguments to the just target.

And I'm concerned that this didn't cause the workflow to fail :confused:.

Reply:
oh, I only just saw this comment.  Will look into that because it sure seems like a problem.

Reply:
Trying a fix in https://github.com/githedgehog/dataplane/pull/294

Reply:
Now merged